deeplearning4j-modelexport-solr/src/test/java/org/deeplearning4j/nn/modelexport/solr/ltr/model/ScoringModelTest.java

```diff
@@ -0,0 +1,225 @@
+import org.junit.Test;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.factory.Nd4j;
+
+public class ScoringModelTest {
+
+  protected static class DummyFeature extends Feature {
+
+    public DummyFeature(String name) {
+      super(name, Collections.EMPTY_MAP);
+    }
+
+    @Override
+    protected void validate() throws FeatureException {
+    }
+
+    @Override
+    public FeatureWeight createWeight(IndexSearcher searcher, boolean needsScores, SolrQueryRequest request,
+        Query originalQuery, Map<String,String[]> efi) throws IOException {
+      return null;
+    }
+
+    @Override
+    public LinkedHashMap<String,Object> paramsToMap() {
+      return null;
+    }
+
+  };
+
+  protected List<Feature> featuresList(int numFeatures) throws Exception {
+    final ArrayList<Feature> features = new ArrayList<Feature>();
+    for (int ii=1; ii<=numFeatures; ++ii)
+    {
+      features.add(new DummyFeature("dummy"+ii));
+    }
+    return features;
+  }
+
+  protected List<Normalizer> normalizersList(int numNormalizers) throws Exception {
+    final ArrayList<Normalizer> normalizers = new ArrayList<Normalizer>();
+    for (int ii=1; ii<=numNormalizers; ++ii)
+    {
+      normalizers.add(IdentityNormalizer.INSTANCE);
+    }
+    return normalizers;
+  }
+
+  protected List<float[]> floatsList(int numFloats) {
+    final List<float[]> floatsList = new ArrayList<float[]>();
+    final float[] floats0 = new float[numFloats];
+    final float[] floats1 = new float[numFloats];
+    for (int ii=0; ii<numFloats; ++ii) {
+      floats0[ii] = 0f;
+      floats1[ii] = 1f;
+    }
+    floatsList.add(floats0);
+    floatsList.add(floats1);
+    return floatsList;
+  }
+
+  @Test
+  public void test() throws Exception {
+    for (int numFeatures = 3; numFeatures <= 5; ++numFeatures) {
+
+      for (Model model : new Model[]{
+          buildMultiLayerNetworkModel(numFeatures),
+          buildComputationGraphModel(numFeatures)
+        }) {
+
+        doTest(model, numFeatures);
+
+      }
+    }
+  }
+
+  private void doTest(Model originalModel, int numFeatures) throws Exception {
+
+    final Path tempDirPath = Files.createTempDirectory(null);
+    final File tempDirFile = tempDirPath.toFile();
+    tempDirFile.deleteOnExit();
+
+    final SolrResourceLoader solrResourceLoader = new SolrResourceLoader(tempDirPath);
+
+    final File tempFile = File.createTempFile("prefix", "suffix", tempDirFile);
+    tempFile.deleteOnExit();
+
+    final String serializedModelFileName = tempFile.getPath();
+
+    ModelSerializer.writeModel(originalModel, serializedModelFileName, false);
+
+    final Model restoredModel = ModelGuesser.loadModelGuess(serializedModelFileName);
+
+    final ScoringModel ltrModel = new ScoringModel(
+        "myModel", featuresList(numFeatures), normalizersList(numFeatures), null, null, null);
+    ltrModel.setSerializedModelFileName(serializedModelFileName);
+    ltrModel.init(solrResourceLoader);
```

Comment:
And once it builds, it will be interesting to see if the tests pass. Previously locally tests passed for me but now am getting this failure here:

```
...
Warning: Loader.putMemberOffset(): java.lang.ClassNotFoundException: org.nd4j.nativeblas.Nd4jCpu$float_sru_old
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.758 sec <<< FAILURE! - in org.deeplearning4j.nn.modelexport.solr.ltr.model.ScoringModelTest
test(org.deeplearning4j.nn.modelexport.solr.ltr.model.ScoringModelTest)  Time elapsed: 1.692 sec  <<< ERROR!
org.apache.solr.ltr.model.ModelException: score(...) test failed for model myModel
	at org.deeplearning4j.nn.modelexport.solr.ltr.model.ScoringModelTest.doTest(ScoringModelTest.java:131)
	at org.deeplearning4j.nn.modelexport.solr.ltr.model.ScoringModelTest.test(ScoringModelTest.java:105)
Caused by: org.nd4j.linalg.workspace.ND4JWorkspaceException: Expected no workspace active in outputOfLayerDetached - Open/active workspaces: [WS_LAYER_WORKING_MEM, WS_LAYER_ACT_0, WS_LAYER_ACT_1]
	at org.deeplearning4j.nn.modelexport.solr.ltr.model.ScoringModelTest.doTest(ScoringModelTest.java:131)
	at org.deeplearning4j.nn.modelexport.solr.ltr.model.ScoringModelTest.test(ScoringModelTest.java:105)


Results :

Tests in error: 
  ScoringModelTest.test:105->doTest:131 » Model score(...) test failed for model...

Tests run: 1, Failures: 0, Errors: 1, Skipped: 0
```

Reply:
You can ignore this: ```Warning: Loader.putMemberOffset(): java.lang.ClassNotFoundException: org.nd4j.nativeblas.Nd4jCpu$float_sru_old```
It's a known issues with the snapshots, shouldn't cause any problems.

As for the exception: it may not be related to your changes/additions here, but I'm not sure yet (this is due to some recent refactoring around workspaces). I'll pull this PR and run it locally to see what's up, and will get back to you.
```
Caused by: org.nd4j.linalg.workspace.ND4JWorkspaceException: Expected no workspace active in outputOfLayerDetached - Open/active workspaces: [WS_LAYER_WORKING_MEM, WS_LAYER_ACT_0, WS_LAYER_ACT_1]
```

Reply:
OK, so yes - that error was an issue on our side.
I've fixed it here: https://github.com/deeplearning4j/deeplearning4j/pull/5047
After rebasing your branch based on that, I'm seeing the test pass. But feel free to confirm that on your end too.